packages/suite/src/components/suite/Settings/components/Row.tsx

```diff
@@ -16,14 +16,10 @@ const Wrapper = styled.div`
     }
 `;
 
-interface Props extends React.HTMLAttributes<HTMLDivElement> {
-    children: React.ReactNode;
-}
+type RowProps = React.HTMLAttributes<HTMLDivElement>;
 
-const Row = ({ children, ...rest }: Props) => (
+export const Row: React.FC<RowProps> = ({ children, ...rest }) => (
```

Comment:
but it actually makes sense here because it had `children` defined, no?

Reply:
it is but, you actually need to know that, if you use standart PropsType, then you can't read that type like a docs when you open this file and you don't need know that there is children type hidden somewhere or even search code if FC is used.

Reply:
Explicit over implicit

Reply:
Not fight, great discussion if it's sensible. It helps both sides get better knowledge of topic, at least for me it works that way because it force me to study topic deeply.

About what you mentioned, core point functional components is that they are plain functions and not some kind of special component functions. There is nothing like React component, there are just functions. You can have something like this:

```jsx
const giveMeNumber = () => 42
...
<giveMeNumber />
```

And this is both plain function and also "React functional component". I don't saying we should start doing that but I want to show there is no difference between function and component.

Overtyping is bad thing and that is exactly what `React.FC` is. It was intended to use only in cases where plain function is not enough and you need to enrich it with some properties like `displayName` etc., that you need very rarely. If you use it everywhere it won't give you any benefit (like return type guard), because if you for example do something `const MyComponent = () => {hello: 'world'}` it will throw in place where component is used because React itself it strongly typed. So it's better to let React decide which function is correct component and which not.

It already proved as wise solution to not use FC everywhere because of breaking changes in latest React 18, where implicit children is removed from that type. It will break code everywhere where you rely on this behavior. And that's result of overtyping, it's less safe again breaking changes. >>> https://twitter.com/mpocock1/status/1512454673034358785